### ISSUE TITLE
[Feature request] Label and barcode for location 

### ISSUE THREAD

**Issue by Ajtak** [grokability/snipe-it#9660]
**Server (please complete the following information):**
- Snipe-IT Version: v5.1.5 - build 6055 (master)
- OS: [e.g. Ubuntu, CentOS]: Ubuntu
- Web Server: [e.g. Apache, IIS]: Apache
- PHP Version: 7.4.3

**Is your feature request related to a problem? Please describe.**
We have many locations in our company, and we would welcome it possible to generate a label with a QR code and barcode for this location. 
**Describe the solution you'd like**
Add field some as Asset tag to location, and add option for generate label.

**Describe alternatives you've considered**

**Additional context**

**Comment by snipe** [comment]
I'm not clear on what this barcode for location would do, exactly? When you scan it, what happens?

**Comment by Ajtak** [comment]
I think that all assets, user, ... for locations.
For example: https://demo.snipeitapp.com/locations/7

<blockquote><img src="https://demo.snipeitapp.com/favicon.ico" width="48" align="right"><div><strong><a href="https://demo.snipeitapp.com/locations/7">Snipe-IT Asset Management Demo</a></strong></div></blockquote>

**Comment by blathers123** [comment]
@Ajtak could you describe your workflow for the qr code?

I would also be interested in this. I have several boxes/bins at different locations. I'd like a QR code on each box that will tell me the contents of the box and where it belongs.

 I had something like this working at one point by adding the boxes and checking assets in and out of the boxes, but the workflow felt tedious and Jacky

**Comment by ancker010** [comment]
Ping for this feature as well.

Like @blathers123, I would label boxes, shelves, tubs, drawers, etc.
I could scan a label and quickly know what's in that location.

**Comment by Oninaig** [comment]
Any update on this or any way to get Snipe-It to support barcodes for "containers" like boxes/shelves that have multiple assets within them? You would scan the barcode for the box/shelf and it would display all of the assets linked to that location

**Comment by mach7** [comment]
@snipe I agree with @ancker010 and @Oninaig That is exactly how I would be using it as well. Be able to scan a work station and seeing the equipment located at that location, or scanning a shelf to see what containers are on it and what contents are in that container. 

So, is there any update on a possible workflow implementation for these situations?

**Comment by snipe** [comment]
Welp, so I've been thinking about this feature for some time. The challenge here is that different people will very likely want different workflows. Some might want to checkin a whole location (or container) via barcode, some might just want a list, and some might want something else entirely. 

> That is exactly how I would be using it as well. Be able to scan a work station

That would imply a workstation is listed as a location, which seems a little weird to me. (People use Snipe-IT in a lot of weird ways though, so...)

I could see the proposed solution really bloating out the locations section and making it difficult to use. 

I've already been working on the concept of "containers", as I'm about to move across the world, and I'll NEED to provide a printed inventory of everything that's in every box in order to clear customs. What I'm trying to decide is if it would be better to use locations in this way, or come up with a second-class item like "containers". This would leave Locations as-is, and would be more useful in the context of, for example, hackerspaces, where a "container" might be "a box or gear temporarily checked out to Hackspace Member XYZ), and the person handling the inventory at the hackerspace could scan the code on the bin, do a visual audit that everything that's supposed to be there is there, and then either through the UI or through a second barcode, scan all of the contents in.

**Comment by mach7** [comment]
What about building it with the following hierarchy:

Warehouse
- Shelving, Lot, Position, Home (All suggested names) 
--- Container, box, bin, holder
----- Asset

Something like that..

**Comment by snipe** [comment]
In this example, is Warehouse a location? 

The problem is that lots of people use us in very different ways, and lots of folks won't have a warehouse.

And in the hackerspace scenario, the "warehouse" might be the facility itself, but they don't necessarily need to know the checkout location. "User XYZ's house" as a location would be a lot of manual overhead.

**Comment by mach7** [comment]
Location:
- Rack
--- Shelf
------ Container
--------- Kit
------------ Components
------------ Consumables
------------ Accessories

**Comment by ancker010** [comment]
I like the container as a separate object from a location.
For me, the container is IN the location. And it's possible that I could move an entire container to a new location.

**Comment by smcpeck** [comment]
Throwing my hat in this ring, too. Would love to be able to scan a bin and see a list of assets/components are in it.

For now, I have a hierarchy of locations like:
```
Rack 1:
  Shelf 1:
    - Bin 1
    - Bin 2
    - Bin 3
```

But I think a moveable container object like @ancker010 suggests would be even better.

My question then becomes: Are racks and shelves still `locations` and the bin on those shelves are these new `container` objects that I can rearrange?  (Really just thinking aloud here.)

**Comment by mach7** [comment]
I just had a thought on this today, would it help to accommodate multiple workflows if there were a new settings page made up to select the hierarchy of the individuals setup?

Something like a few drop down selections where you select:

Top level
- sub-level 1
-- micro-level 1
-- micro-level 2
- sub-level 2
-- micro-level 1
-- micro-level 2

Give the barcode the ability to work with any one of them and then it would just be up to each user to setup their own hierarchy. Would this end up being beneficial?

**Comment by mikeatals** [comment]
Jumping into this as well.

A 2nd tier location field would be nice.  

For me I just want the office/cube number.  I really like knowing all my computers in a building through the location tab as it is now.  I just wish I could assign them to a physical room.  


I'm not asking for this, but this was a workflow we had in a different system...
auditor would scan the barcode on the cubical, then scan all the items they can find in the cubical, end of day they'd have a report on what wasn't found, was moved, ect.

**Comment by maddin8** [comment]
I would also love to see this, my use case would be to have various larger storage boxes as locations which then contain assets. Scanning a barcode that the storage boxes are labelled with would help quickly identifying whats in the box without looking into it (the barcode takes me to the location in Snipe-IT and shows me all related objects).

**Comment by Pugglewuggle** [comment]
Bump. This looks useful for store rooms of equipment.

**Comment by snipe** [comment]
Best we can do for now is this: https://github.com/snipe/snipe-it/pull/16651

If we need to create entirely separate labels for locations, that will complicate the label maker pretty dramatically.
